functions/src/main/java/org/neo4j/doc/FunctionDescriptionsGenerator.java

```diff
@@ -0,0 +1,254 @@
+    {
+        Map<Category,List<FunctionDescription>> functions;
+        try ( Transaction tx = db.beginTx() )
+        {
+            String query = "CALL dbms.functions()";
+            try ( Result result = tx.execute( query ) )
+            {
+                functions = parseResult( result );
+            }
+        }
+        return functions;
+    }
+
+    private Map<Category,List<FunctionDescription>> parseResult( Result result )
+    {
+        Map<Category,List<FunctionDescription>> functions = new HashMap<>();
+        result.stream().forEach( row ->
+                                 {
+                                     FunctionDescription p = new FunctionDescription( row );
+                                     if ( !functions.containsKey( p.category() ) )
+                                     {
+                                         functions.put( p.category(), new ArrayList<>() );
+                                     }
+                                     functions.get( p.category() ).add( p );
+                                 } );
+
+        return functions;
+    }
+
+    static class FunctionDescription implements Comparable<FunctionDescription>
+    {
+        private String name;
+        private final String description;
+        private final Category category;
+        private final String signature;
+
+        FunctionDescription( Map<String,Object> row )
+        {
+            setName( (String) row.get( "name" ) );
+            this.description = (String) row.get( "description" );
+            this.category = Category.parse( (String) row.get( "category" ), this.name, this.description );
+            this.signature = (String) row.get( "signature" );
+        }
+
+        void setName( String name )
+        {
+            this.name = name.endsWith( "()" ) ? name : name + "()";
+        }
+
+        Category category()
+        {
+            return category;
+        }
+
+        String signature()
+        {
+            return signature;
+        }
+
+        @Override
+        public boolean equals( Object o )
+        {
+            if ( this == o )
+            {
+                return true;
+            }
+            if ( o == null || getClass() != o.getClass() )
+            {
+                return false;
+            }
+
+            FunctionDescription procedure = (FunctionDescription) o;
+
+            if ( !name.equals( procedure.name ) )
+            {
+                return false;
+            }
+            return signature.equals( procedure.signature );
+        }
+
+        @Override
+        public int hashCode()
+        {
+            int result = name.hashCode();
+            result = 31 * result + signature.hashCode();
+            return result;
+        }
+
+        public String row()
+        {
+            return "| <<functions-" + name.toLowerCase().replace( "()", "" ) + "," + name + ">>  | " + description + "\n";
```

Comment:
I am not sure if this is a problem but if running on windows the `\n` might cause issues, maybe use `System.lineBreak()` instead.